### PR TITLE
Add JWT authentification support

### PR DESCRIPTION
**Pull request by Jhobahtes**
Add an endpoint to generate a JWT token, based on a password flow and Hyperion database
Add a dependency to check validity of the Bearer header

### Usage

Install the new requirements
```bash
pip install -r requirements.txt
```

1. Create a new user (currently without authentication)
```bash
curl --location --request POST 'http://127.0.0.1:8000/users' \
--header 'Content-Type: application/json' \
--data-raw '{
    "login": "jhobahtes",
    "password": "azerty",
    "name": "jhobahtes",
    "firstname": "jhobahtes",
    "nick": "jhobahtes",
    "birth": "01012000",
    "promo": "E",
    "floor": "A",
    "email": "jhobahtesemail",
    "created_on": 1
}'
```

2. Get a JWT for this user
```bash
curl --location --request POST 'http://127.0.0.1:8000/auth/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'username=jhobahtes' \
--data-urlencode 'password=azerty'
```

3. Show existing users (Bearer authentication required, use the token you got during step 2)
```bash
curl --location --request GET 'http://127.0.0.1:8000/users/' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJqaG9iYWh0ZXMiLCJleHAiOjE2NDkwMTcyNjF9.fSmvjIyxuchsVeVHCLyeTvmhcJGLjaOS07D5b2xBxW4'
```

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the documentation, if necessary

### CODE REVIEW

File: app/dependencies.py [hunk 1]
```diff
@@ -1,10 +1,42 @@
...
+
+from app.schemas import schemas_tokens
+from app.core import security
+from app.core.settings import settings
+from app.cruds import cruds_users
+from app.models import models_users
+
 from app.database import SessionLocal
 
+reusable_oauth2 = OAuth2PasswordBearer(tokenUrl="/auth/token")
```
Comment (Victor-Angot) [inline app/dependencies.py:15]: Why do you call the schema reusable ?

File: app/dependencies.py [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
     db = SessionLocal()
     try:
         yield db
     finally:
         db.close()
+
+
+def get_current_user(
+    db: Session = Depends(get_db), token: str = Depends(reusable_oauth2)
+) -> models_users.CoreUser:
```
Comment (Victor-Angot) [inline app/dependencies.py:28]: Could you comment this syntax ?

File: app/dependencies.py (hunk 1 above)
Comment (Jhobahtes) [reply app/dependencies.py:15]: It's the name used by FastAPI.
https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/

**Comment by Jhobahtes** [comment]
Superseded by https://github.com/aeecleclair/Hyperion/pull/8

**Pull request closed without merging**